src/test/kotlin/tech/relaycorp/relaynet/cogrpc/client/CogRPCClientTest.kt

```diff
@@ -333,8 +334,10 @@ internal class CogRPCClientTest {
         }
     }
 
-    private fun buildAndStartServer(service: BindableService) {
-        testServer = TestCogRPCServer(HOST, PORT, service).apply { start() }
+    private fun buildAndStartServer(service: BindableService): String {
+        val port = Random.nextInt(8000, 19999) // Avoid server overlap in parallel tests
```

Comment:
This would make the test flaky (it'd fail if the port is taken). If we do need to use random ports, I'd rather use one we know to be free. E.g.,

```kotlin
// Untested. Inspired from https://stackoverflow.com/questions/2675362/how-to-find-an-available-port

fun findFreePort(): Int {
    val socket = new ServerSocket(0)
    val port = socket.getLocalPort()
    socket.close()
    return port
}
```